docs/notebooks/active_learning_for_binary_classification.pct.py

```diff
@@ -0,0 +1,191 @@
+
+import trieste
+from trieste.acquisition.function import BayesianActiveLearningByDisagreement
+from trieste.acquisition.rule import OBJECTIVE
+from trieste.models.gpflow.models import VariationalGaussianProcess
+from trieste.objectives.utils import mk_observer
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## The problem
+
+# %% [markdown]
+# In Trieste, it is also possible to query most interesting points for learning the problem, i.e we want to have as little data as possible to construct the best possible model (active learning). In this tutorial we will try to do active learning for binary classification problem using Bayesain Active Learning by Disagreement (BALD) for a Gaussian Process Classification Model.
+#
+# Our problem is classification problem of circle dataset. In this toturial we assume that the input space is continous so we can use continuous optimiser for our BALD acquisition function.
+
+# %%
+search_space = trieste.space.Box([-1, -1], [1, 1])
+input_dim = 2
+
+
+def circle(x):
+    return tf.cast((tf.reduce_sum(tf.square(x), axis=1, keepdims=True) - 0.5) > 0, tf.float64)
+
+
+# %%
+density = 100
+xx = np.linspace(search_space.lower[0], search_space.upper[0], density)
+
+grid_xx = np.meshgrid(*[xx] * input_dim)
+xx = np.vstack([g.ravel() for g in grid_xx]).T
+yy = circle(xx).numpy()
+
+plt.figure(figsize=(5, 5))
+plt.contour(*grid_xx, np.reshape(yy, [density] * input_dim), levels=[0.5])
+idx = np.squeeze(yy).astype(bool)
+plt.scatter(xx[idx][:, 0], xx[idx][:, 1], label="1")
+plt.scatter(xx[np.logical_not(idx)][:, 0], xx[np.logical_not(idx)][:, 1], label="0")
+plt.legend()
+plt.show()
+
+# %% [markdown]
+# Let's generate some data for our initial model. Here we randomly sample 10 data points.
+
+# %%
+numSamples = 10
+X = search_space.sample(numSamples)
+observer = mk_observer(circle, OBJECTIVE)
+datasets = observer(X)
+
+# %% [markdown]
+# ## Modelling the binary classification task
+
+# %% [markdown]
+# For the binary classification model, we use the Variational Gaussian Process with Bernoulli likelihood. For more detail of this model, see <cite data-cite="Nickisch08a">[Nickisch et al.](https://www.jmlr.org/papers/volume9/nickisch08a/nickisch08a.pdf)</cite>.
+
+# %%
+from trieste.models.gpflow import GPflowModelConfig, VariationalGaussianProcess
+
+
+def create_bo_model(data):
+    kernel = gpflow.kernels.SquaredExponential(lengthscales=[0.2, 0.2])
+    m = gpflow.models.VGP(data.astuple(), likelihood=gpflow.likelihoods.Bernoulli(), kernel=kernel)
+    return trieste.models.create_model(
+        GPflowModelConfig(
+            **{
+                "model": m,
+                "optimizer": gpflow.optimizers.Scipy(),
+                "optimizer_args": {
+                    "minimize_args": {"options": dict(maxiter=100.0)},
+                },
+            }
+        )
+    )
+
+
+"""
+# model seems harder to train using default VariationalGaussianProcess() optimiser which is adam?
```

Comment:
no, the default is Scipy, if you want to use natgrads then you need to set up Adam with `BatchOptimizer` wrapper, try setting up that and check the performance

Reply:
Fixed the performance by setting max_iter to 100, by still using Scipy optimiser